src/main/java/technology/tabula/CohenSutherlandClipping.java

```diff
@@ -34,100 +34,108 @@
             }
-            else if ((outsidePointRegion & RIGHT) != INSIDE) {
-                outsidePointX = xMax;
-                outsidePointY = delta(outsidePointX, point1X)*lineSlope + point1Y;
+            else if ((outsidePoint.region & RIGHT) != INSIDE) {
+                outsidePoint.x = xMax;
+                outsidePoint.y = delta(outsidePoint.x, point1.x)*lineSlope + point1.y;
             }
-            else if ((outsidePointRegion & BOTTOM) != INSIDE) {
-                outsidePointY = yMin;
-                outsidePointX = lineIsVertical
-                    ? point1X
-                    : delta(outsidePointY, point1Y)/lineSlope + point1X;
+            else if ((outsidePoint.region & BOTTOM) != INSIDE) {
+                outsidePoint.y = yMin;
+                outsidePoint.x = lineIsVertical
+                    ? point1.x
+                    : delta(outsidePoint.y, point1.y)/lineSlope + point1.x;
             }
-            else if ((outsidePointRegion & TOP) != INSIDE) {
-                outsidePointY = yMax;
-                outsidePointX = lineIsVertical
-                    ? point1X
-                    : delta(outsidePointY, point1Y)/lineSlope + point1X;
+            else if ((outsidePoint.region & TOP) != INSIDE) {
+                outsidePoint.y = yMax;
+                outsidePoint.x = lineIsVertical
+                    ? point1.x
+                    : delta(outsidePoint.y, point1.y)/lineSlope + point1.x;
             }
 
-            if (outsidePointRegion == point1Region) {
-                point1X = outsidePointX;
-                point1Y = outsidePointY;
-                point1Region  = regionCode(point1X, point1Y);
+            if (outsidePoint.isInTheSameRegionAs(point1)) {
+                point1.setPositionAndRegion(outsidePoint.x, outsidePoint.y);
             }
             else {
-                point2X = outsidePointX;
-                point2Y = outsidePointY;
-                point2Region = regionCode(point2X, point2Y);
+                point2.setPositionAndRegion(outsidePoint.x, outsidePoint.y);
             }
         }
-        line.setLine(point1X, point1Y, point2X, point2Y);
+        line.setLine(point1.x, point1.y, point2.x, point2.y);
         return true;
     }
 
+    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - //
     private static double delta(double value1, double value2) {
         return (Math.abs(value1 - value2) < MINIMUM_DELTA) ? 0 : (value1 - value2);
     }
 
+    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - //
+    class Point {
+        double x, y;
+        int region;
+
+        Point(double x, double y) {
+            setPositionAndRegion(x, y);
+        }
+
+        void setPositionAndRegion(double x, double y) {
+            this.x = x; this.y = y;
+            region = (x < xMin) ? LEFT : (x > xMax) ? RIGHT : INSIDE;
+            if (y < yMin)
+                region |= BOTTOM;
```

Comment:
I think that in this part the operations are reversed. The logical OR should be done with the constant TOP. See [here](https://en.wikipedia.org/wiki/Cohen%E2%80%93Sutherland_algorithm) the bits table.